### PR TITLE
[55628] iOS: add extra methods to app delegate to detect life cycling of views

### PR DESCRIPTION
**Pull request by MIKKH**

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs [hunk 1]
```diff
@@ -44,20 +44,20 @@ public override void ViewWillAppear(bool animated)
...
+            MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_APP_BECAME_ACTIVE, (object _) =>
             {
                 LogUtils.LogMessage(LogSeverity.INFO, "The user is seeing Countries Consent", null, GetCorrelationId());
             });
 
-            MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_APP_WILL_ENTER_BACKGROUND, (object _) =>
+            MessagingCenter.Subscribe<object>(this, MessagingCenterKeys.KEY_APP_RESIGN_ACTIVE, (object _) =>
             {
-                LogUtils.LogMessage(LogSeverity.INFO, "The user left Countries Consent", null, GetCorrelationId());
+                LogUtils.LogMessage(LogSeverity.INFO, "The user is left Countries Consent", null, GetCorrelationId());
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs:60]: Looks like your are bringing the typo back?

File: NDB.Covid19/NDB.Covid19/Utils/MessagingCenterKeys.cs [hunk 2]
```diff
@@ -12,5 +12,8 @@ public static class MessagingCenterKeys
         public static string KEY_PERMISSIONS_CHANGED => "PermissionsChangedKey";
 
         public static string KEY_UPDATE_DAILY_NUMBERS => "KEY_UPDATE_DAILY_NUMBERS";
+
+        public static string KEY_APP_BECAME_ACTIVE => "KEY_APP_BECAME_ACTIVE";
```
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Utils/MessagingCenterKeys.cs:16]: Maybe delete old keys (e.g., KEY_APP_RETURNS_FROM_BACKGROUND and KEY_APP_WILL_ENTER_BACKGROUND if they are no longer used?)

**Review by pavshr — CHANGES_REQUESTED** [review]

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs (hunk 1 above)
Comment (MIKKH) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs]: I have corrected the error.

File: NDB.Covid19/NDB.Covid19/Utils/MessagingCenterKeys.cs (hunk 2 above)
Comment (MIKKH) [reply NDB.Covid19/NDB.Covid19/Utils/MessagingCenterKeys.cs:16]: The old key is still used in InfectionStatus.
It is unclear to me if we can change the monitored event here to app becomes active/inactive.

I took a look at the implementation in the Danish app, and they use the "old" keys in InfectionStatus.

Comment (pavshr) [reply NDB.Covid19/NDB.Covid19/Utils/MessagingCenterKeys.cs:16]: correct :) thanks

**Review by pavshr — APPROVED** [review]

**Pull request merged**